2_process/src/training_data.py

```diff
@@ -0,0 +1,165 @@
+    where_to_split = np.insert(where_to_split, 0, 0)
+    # The first split will be where_to_split[0]:where_to_split[1]
+    # The second split will be where_to_split[1]:where_to_split[2]
+    # And so on.
+
+    # Split item_indices, using where_to_split to slice
+    split_indices = [item_indices[where_to_split[i_split]:where_to_split[i_split+1]] 
+                     for i_split in range(num_splits)]
+    return split_indices
+
+
+def read_sequences(lake_sequence_files, n_depths):
+    """
+    Read a list of sequence files in, remove sequences that have any
+    NaN values among their inputs, and return a single numpy array
+    containing all sequences from all sequence files.
+
+    :param lake_sequence_files: List of lake sequence .npy files
+    :param n_depths: Number of discrete depths at which LSTM will
+    output temperatures
+    :returns: numpy array of sequences
+
+    """
+    sequences = np.concatenate([np.load(sfn) for sfn in lake_sequence_files])
+    # remove inputs with nans
+    # shape of sequences is (# sequences, sequence_length, # depths + # features)
+    # slice from n_depths forward to index features
+    nan_inputs = np.any(np.isnan(sequences[:, :, n_depths:]), axis=(1,2))
+    return sequences[np.invert(nan_inputs), :, :]
+
+
+def get_mean_std(data, **kwargs):
+    """
+    Get means and standard deviations of an array of data. Ignore NaN
+    values. Replace any standard deviations of 0 with 1.
+
+    :param data: Array-like set of data
+    :param **kwargs: Keyword arguments used by np.nanmean and np.nanstd
+
+    """
+    means = np.nanmean(data, **kwargs)
+    stds = np.nanstd(data, **kwargs)
+    # Correct for any 0 standard deviations.
+    stds = np.where(stds <= 0, 1, stds)
+    return (means, stds)
+
+
+def get_train_test_data(sequences_summary_file,
+                        train_frac,
+                        test_frac,
+                        n_depths,
+                        n_dynamic,
+                        n_static,
+                        seed):
+    """
+    Create train and test data
+
+    This function takes data from lake-specific sequences in .npy files and creates arrays for training and testing.
+
+    1. Split lakes into train and test sets
+    2. Read lake-specific sequences from .npy files and concatenate
+    3. Standardize using training data
+
+    """
+
+    # 1. Split lakes into train and test sets
+    sequences_summary = pd.read_csv(sequences_summary_file)
+
+    # Remove lakes with 0 sequences
+    nonzero_sequences = sequences_summary[sequences_summary.num_sequences>0]
+    num_lakes = len(nonzero_sequences)
+
+    # Split lake indices into train and test
+    train_lakes, test_lakes = split_indices(num_lakes, [train_frac, test_frac], seed=seed)
+
+    lake_sequences_files_train = nonzero_sequences.sequences_file.iloc[train_lakes]
+    lake_sequences_files_test = nonzero_sequences.sequences_file.iloc[test_lakes]
```

Comment:
I might just be missing this, but are we retaining this information about which files fall into each group anywhere? I seems like it would be good to have a record of which lakes will be used for training and which will be used for testing.  I could see storing either these file names, or extracting the site_ids from the file names and storing those.

Reply:
You're right! We'll want to know which lakes were used for training and testing. I think another summary file would be useful for saving that info.